ivy_tests/test_ivy/test_frontends/test_torch/test_spectra_ops.py

```diff
@@ -0,0 +1,1620 @@
+                    dtype=dtype,
+                    shape=shape,
+                    abs_smallest_val=abs_smallest_val,
+                    min_value=min_value,
+                    max_value=max_value,
+                    allow_inf=allow_inf,
+                    exclude_min=exclude_min,
+                    exclude_max=exclude_max,
+                    large_abs_safety_factor=large_abs_safety_factor,
+                    small_abs_safety_factor=small_abs_safety_factor,
+                    safety_factor_scale=safety_factor_scale,
+                )
+            )
+        )
+
+    value1 = values[0]
+
+    correction = draw(helpers.ints(min_value=0, max_value=1))
+
+    fweights = draw(
+        helpers.array_values(
+            dtype="int64",
+            shape=shape[1],
+            abs_smallest_val=1,
+            min_value=1,
+            max_value=10,
+            allow_inf=False,
+        )
+    )
+
+    aweights = draw(
+        helpers.array_values(
+            dtype="float64",
+            shape=shape[1],
+            abs_smallest_val=1,
+            min_value=1,
+            max_value=10,
+            allow_inf=False,
+            small_abs_safety_factor=1,
+        )
+    )
+
+    return [dtype], value1, correction, fweights, aweights
+
+
+# cov
+@handle_frontend_test(
+    fn_tree="torch.cov",
+    dtype_x1_corr_cov=_get_dtype_value1_value2_cov(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=2,
+        max_dim_size=5,
+        min_value=1,
+        max_value=1e10,
+        abs_smallest_val=0.01,
+        large_abs_safety_factor=2,
+        safety_factor_scale="log",
+    ),
+    test_with_out=st.just(False),
+)
+def test_torch_cov(
+    dtype_x1_corr_cov,
+    test_flags,
+    frontend,
+    fn_tree,
+    on_device,
+    backend_fw,
+):
+    dtype, x1, correction, fweights, aweights = dtype_x1_corr_cov
+    helpers.test_frontend_function(
+        input_dtypes=["float64", "int64", "float64"],
+        backend_to_test=backend_fw,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        rtol=1e-2,
+        atol=1e-2,
+        input=x1,
+        correction=correction,
+        fweights=fweights,
+        aweights=aweights,
+    )
+
+
+# view_as_real
+@handle_frontend_test(
+    fn_tree="torch.view_as_real",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        num_arrays=1,
+    ),
+)
+def test_torch_view_as_real(
+    *,
+    dtype_and_x,
+    on_device,
+    fn_tree,
+    frontend,
+    test_flags,
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=np.asarray(x[0], dtype=input_dtype[0]),
+    )
+
+
+# stft
+@handle_frontend_test(
```

Comment:
Same for this, we only need this test.

Reply:
Good day, @zaeemansari70 ,

I have made the requested changes. Thank you.
However, I have further questions;
1. >>
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-6-6b3ffb48f7e4>](https://localhost:8080/#) in <cell line: 5>()
      3 
      4 import numpy as np
----> 5 from hypothesis import assume, strategies as st
      6 import hypothesis.extra.numpy as nph
      7 

ModuleNotFoundError: No module named 'hypothesis'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------
 How can I find the issue it has?

2.>> How can I find a compiled list of ivy function types and their tests codes. The document has some few examples like ivy.tan( ), ivy.tan( )

Thank you.

Reply:
You have not properly configured the environment, I'd suggest follow this according to your use case https://unify.ai/docs/ivy/overview/contributing/setting_up.html Thanks!